object.go

```diff
@@ -434,18 +432,21 @@ func (obs *obs) Put(meta *ObjectMeta, r io.Reader, opts ...ObjectOpt) (*ObjectIn
 	select {
 	case <-js.PublishAsyncComplete():
 		if err := getErr(); err != nil {
-			purgePartial()
+			if r != nil {
+				purgePartial()
+			}
 			return nil, err
 		}
 	case <-time.After(obs.js.opts.wait):
 		return nil, ErrTimeout
 	}
-	info.ModTime = time.Now().UTC()
 
-	// Delete any original one.
+	info.ModTime = time.Now().UTC() // TODO get the time from the PublishAck
```

Comment:
can someone help me with this? The PublishAck has the message time, I just don't know how to get it.